pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -139,6 +139,12 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 
 	// Don't modify the informer's copy.
 	pr := original.DeepCopy()
+
+	if pr.IsCleanedUp() {
+		c.Logger.Info("Not reconciling cleaned up PipelineRun")
```

Comment:
Similarly to my comment on taskruns, I suggest including the PipelineRun's name and the name of the specific condition that is leading to the skip:

```go
c.Logger.Infof("Skipping reconcile of PipelineRun %q: condition %s is True", tr.Name, v1beta1.ConditionCleanedUp)
```

Reply:
IIUC this will happen in a few cases:
- the updates done during `isDone` (including setting this condition) will trigger the informer and get this key re-queued
- the periodic re-sync 
- a restart of the controller

The last two may take a while when there are a lot of items in etcd, so it's good to stop reconcile as early as possible :)